tests/components/hyperion/test_light.py

```diff
@@ -420,11 +622,26 @@ async def test_full_state_loaded_on_start(hass):
     }
     client.effects = [{const.KEY_NAME: "One"}, {const.KEY_NAME: "Two"}]
 
-    await setup_entity(hass, client=client)
+    await setup_test_config_entry(hass, client=client)
 
-    entity_state = hass.states.get(TEST_ENTITY_ID)
+    entity_state = hass.states.get(TEST_ENTITY_ID_1)
 
     assert entity_state.attributes["brightness"] == round(255 * (brightness / 100.0))
     assert entity_state.attributes["effect"] == hyperion_light.KEY_EFFECT_SOLID
     assert entity_state.attributes["icon"] == hyperion_light.ICON_LIGHTBULB
     assert entity_state.attributes["hs_color"] == (180.0, 100.0)
+
+
+async def test_unload_entry(hass):
+    """Test unload."""
+    assert DOMAIN not in hass.data
+
+    entry = await setup_test_config_entry(hass)
+    assert hass.states.get(TEST_ENTITY_ID_1) is not None
+    client = hass.data[DOMAIN][entry.entry_id][CONF_ROOT_CLIENT]
+    assert client.async_client_connect.called
+    assert not client.async_client_disconnect.called
+
+    assert await async_unload_entry(hass, entry)
```

Comment:
Only use the core interfaces to start the tests. Here use `hass.config_entries.async_unload`.